test/tests/functional/pbs_passing_environment_variable.py

```diff
@@ -102,4 +102,6 @@ def test_passing_shell_function(self):
         with open(job_outfile, 'r') as f:
             job_output = f.read().strip()
         self.assertEqual(job_output,
-                         'foo="(){if [ /bin/true ];\nthen\necho hello;\nfi};"')
+                         'foo="(){if [ /bin/true ];\nthen\necho hello;\nfi};"',
+                         msg="Environment variable foo content does \
+                         not match original")
```

Comment:
If you want to break a string onto two lines, you need to use string literal concatenation (two quoted strings).  This will literally continue the string onto the next line and you will have all that whitespace in your string.